---
- name: Update sctask deatils in ServiceNow
  hosts: localhost
  gather_facts: true

  vars:
    servicenow_instance:  "https://dev213985.service-now.com" #"{{ servicenow_instance }}"
    pat_token: "KoD9kh3m2FES-_GuH_0eRo4d-Lk4Eu1CNuM_4zclgZ6E6UTtRtTZNxca-8X9iGfgyO0JCkXJj-_0MEXnBbjXRg" #"{{ pat_token }}"
    sys_id: "21c48579932a31108a4470918bba107d" #"{{sys_id}}"
    user_sys_id: "" #"6816f79cc0a8016401c5a33be04be441" #"{{ user_sys_id }}"
    extracted_output: "#03* - Automation In Progress - Testing2" #"{{ extracted_output }}"
    sctask_state: "2" #"{{ sctask_state}}"
        
  tasks:
    - block:
      - name: Update some fields in ServiceNow sctasks
        uri:
          url: "{{ servicenow_instance }}/api/now/table/sc_task/{{ sys_id }}"
          method: PATCH
          body_format: json
          headers:
            Accept: "application/json"
            #Authorization: "Basic {{ encoded_credentials | string }}"
            Authorization: "Bearer {{ pat_token }}"
          return_content: yes
          force_basic_auth: yes
          body:
            state: "{{ sctask_state}}"
            work_notes: "{{ extracted_output }}"
            assigned_to: "{{ user_sys_id }}"

      rescue:
        - name: Log Output For Failure
          debug:
            msg: "Log : FAILED : Task {{ ansible_failed_task.name }} Failed and the ERROR: {{ ansible_failed_result }}"
          failed_when: True

        # - name: set fact ansible_failed_result
        #   set_fact:
        #     ansible_fail_result: "{{ ansible_failed_result.msg  }}"

        # - name: print ansible_fail
        #   debug:
        #     var: ansible_fail_result 

        # - name: call logging playbook 
        #   include_tasks: logging.yml
        #   vars:
        #     ansible_failed_results: "{{ ansible_fail_result  }}"




